crates/wast/src/wast.rs

```diff
@@ -134,9 +141,26 @@ impl WastContext {
 
     /// Register an instance to make it available for performing actions.
     fn register(&mut self, name: Option<&str>, as_name: &str) -> Result<()> {
-        let instance = self.get_instance(name)?.clone();
-        self.linker.instance(as_name, &instance)?;
-        self.instances.insert(as_name.to_string(), instance);
+        match name {
+            Some(name) => {
+                let items = self
+                    .linker
+                    .iter()
+                    .filter(|(module, _, _)| *module == name)
+                    .map(|(_, name, item)| (name.to_string(), item.clone()))
+                    .collect::<Vec<_>>();
+                for (name, item) in items {
+                    self.linker.define(as_name, &name, item)?;
+                }
```

Comment:
Could we factor out this logic too, with something like `self.linker.alias(name, as_name)` or so?